docker/docker_image_src.go

```diff
@@ -39,6 +39,12 @@ func newImageSource(ctx *types.SystemContext, ref dockerReference, requestedMani
 	if requestedManifestMIMETypes == nil {
 		requestedManifestMIMETypes = manifest.DefaultRequestedManifestMIMETypes
 	}
+	for _, mtrequested := range requestedManifestMIMETypes {
+		if !supportedManifestMIMETypesMap()[mtrequested] {
+			requestedManifestMIMETypes = manifest.DefaultRequestedManifestMIMETypes
+			break
```

Comment:
This would ignore the input value if _any_ value is unrecognized; I think it should ignore input only if _all_ values are unrecognized (so that a caller can say “i prefer v2s1, then OCI, then v2s2 as a last resort”.)